components/admin_console/billing/billing_history.tsx

```diff
@@ -185,60 +178,48 @@ const BillingHistory: React.FC<Props> = () => {
-                    const fullUsers = invoice.line_items.filter((item) => item.type === 'full').reduce((val, item) => val + item.quantity, 0);
-                    const partialUsers = invoice.line_items.filter((item) => item.type === 'partial').reduce((val, item) => val + item.quantity, 0);
-
-                    return (
-                        <tr
-                            className='BillingHistory__table-row'
-                            key={invoice.id}
-                        >
-                            <td>
-                                <FormattedDate
-                                    value={new Date(invoice.period_start)}
-                                    month='2-digit'
-                                    day='2-digit'
-                                    year='numeric'
-                                    timeZone='UTC'
-                                />
-                            </td>
-                            <td>
-                                <div>{product?.name}</div>
-                                <div className='BillingHistory__table-bottomDesc'>
-                                    <FormattedMarkdownMessage
-                                        id='admin.billing.history.usersAndRates'
-                                        defaultMessage='{fullUsers} users at full rate, {partialUsers} users with partial charges'
-                                        values={{
-                                            fullUsers,
-                                            partialUsers,
-                                        }}
-                                    />
-                                </div>
-                            </td>
-                            <td className='BillingHistory__table-total'>
-                                <FormattedNumber
-                                    value={(invoice.total / 100.0)}
-                                    // eslint-disable-next-line react/style-prop-object
-                                    style='currency'
-                                    currency='USD'
-                                />
-                            </td>
-                            <td>
-                                {getPaymentStatus(invoice.status)}
-                            </td>
-                            <td className='BillingHistory__table-invoice'>
-                                <a
-                                    target='_new'
-                                    rel='noopener noreferrer'
-                                    href={Client4.getInvoicePdfUrl(invoice.id)}
-                                >
-                                    <i className='icon icon-file-pdf-outline'/>
-                                </a>
-                            </td>
-                        </tr>
-                    );
-                })}
+                {billingHistory.map((invoice: Invoice) => (
```

Comment:
The real changes in this block are separating our the invoice user count into its own component, and reading the invoice plan name from new `current_product_name` instead of inferring it from product ids.